compiler/rustc_middle/src/ty/codec.rs

```diff
@@ -50,22 +50,6 @@ impl<'tcx, E: TyEncoder<'tcx>> EncodableWithShorthand<'tcx, E> for ty::Predicate
     }
 }
 
-pub trait OpaqueEncoder: Encoder {
```

Comment:
Right now, the only use for `OpaqueEncoder` is to workaround a specialization bug which prevents from using `opaque::Encoder` everywhere in the `OnDiskCache`. `OpaqueEncoder` may rather belong to `ty::query::on_disk_cache`, with its use restricted in that module.